### PR TITLE
Change json gtest environment variable to compile-time definition

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Changes the `NJP_DEBUG_DUMP` environment variable used in `json_tree.cpp` gtest source to a local `#define` compile-time definition instead. This preserves the rather complex code for debugging json parsing without requiring building it otherwise.
This is part of larger work to refactor and document env vars in the libcudf code.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/tests/io/json_tree.cpp [hunk 1]
```diff
@@ -882,11 +889,10 @@ TEST_P(JsonTreeTraversalTest, CPUvsGPUTraversal)
                                                      json_lines,
                                                      stream,
                                                      rmm::mr::get_current_device_resource());
-  // Print tree representation
-  if (std::getenv("NJP_DEBUG_DUMP") != nullptr) {
-    printf("AFTER  traversal (gpu_tree):\n");
-    print_tree(gpu_tree);
-  }
+#if NJP_DEBUG_DUMP
+  printf("AFTER  traversal (gpu_tree):\n");
```
[the remark this answers is not in the document]
Comment (davidwendt) [reply cpp/tests/io/json_tree.cpp:893]: I believe this was done on purpose to line up with line 880:  `BEFORE` output (which has 1 more letter than `AFTER`) .

[the remark this answers is not in the document]
Comment (davidwendt) [reply cpp/tests/io/json_tree.cpp:893]: Aesthetics is to be admired not scorned.

**Review by bdice — APPROVED** [review]
Looks fine to me. I would prefer a different name that I recognize -- `NJP` presumably stands for "new JSON parser" but I wouldn't have guessed that without the crowd wisdom from @vuule's recent talk. Suggesting `LIBCUDF_JSON_DEBUG_DUMP`.

File: cpp/tests/io/json_tree.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (vuule) [reply cpp/tests/io/json_tree.cpp:893]: I love that @davidwendt figured it out :)
We should take this opportunity to argue about "tabs vs spaces" once again :D

**Review by karthikeyann — APPROVED** [review]
Change looks good to me. 
Macro`NJP_DEBUG_DUMP` was introduced during development of nested json parser (NJP @elstehle ) and we continued to use it.
Reasoning for still having these macros around: these macros are useful in debugging the json reader while adding new features.

**Pull request merged**